numpy/ma/tests/test_core.py

```diff
@@ -48,6 +48,7 @@
     subtract, sum, take, tan, tanh, transpose, where, zeros,
     )
 from numpy.compat import pickle
+from numpy.polynomial import Polynomial, polynomial
```

Comment:
```suggestion
from numpy.polynomial import Polynomial
```

Reply:
Yes you're right. I tried to use MaskedConstant instead but it seems like I can't do computation upon it. I believe the best fit would be to modify _array_finalize_ function to output the actually masked array, instead of the _data. And I tried nan because if you do mathematical operation like add on nan, the output is still nan. So I guess it kinda mimics the masked array behavior.